### DISCUSSION TITLE
unsupported GNU version! gcc versions later than 6 are not supported!

### DISCUSSION THREAD

**Discussion by abhidipbhattacharyya** [horovod/horovod#3126] [Q&A]
I am trying to install horovod.  At around 66% of installation I am getting error:

In file included from /usr/include/host_config.h:50,
                     from /usr/include/cuda_runtime.h:78,
                     from <command-line>:
    /usr/include/crt/host_config.h:121:2: error: #error -- unsupported GNU version! gcc versions later than 6 are not supported!
      121 | #error -- unsupported GNU version! gcc versions later than 6 are not supported!
          |  ^~~~~
    In file included from /usr/include/driver_types.h:72,
                     from /usr/include/builtin_types.h:59,
                     from /usr/include/cuda_runtime.h:86,
                     from <command-line>:
    /usr/include/limits.h:26:10: fatal error: bits/libc-header-start.h: No such file or directory
       26 | #include <bits/libc-header-start.h>
          |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated.
    CMake Error at horovod_cuda_kernels_generated_cuda_kernels.cu.o.RelWithDebInfo.cmake:220 (message):
      Error generating
      /tmp/pip-install-l0oxred5/horovod_c2af735b5baf4bb78a32f78a8e466319/build/temp.linux-x86_64-3.7/RelWithDebInfo/horovod/common/ops/cuda/CMakeFiles/horovod_cuda_kernels.dir//./horovod_cuda_kernels_generated_cuda_kernels.cu.o

I am using Pop!_OS 18.04 LTS. I can not find any option to give the path to GCC 6 in installation command. Please let me know if more information is required.

**Comment by romerojosh** [comment]
What CUDA toolkit version are you using? You may need to upgrade to support a newer GCC version.

**Reply by abhidipbhattacharyya** [reply]
Hello,
My current default GCC version is 7.5.0. nvidia-smi shows cuda version 11.4.

**Reply by maxhgerlach** [reply]
The CUDA toolkit is versioned separately from the CUDA driver version (not sure about the proper terms here). On Ubuntu I would install something like `cuda-libraries-11-0` for instance to get CUDA 11.0, which would also work with a driver that announces "cuda version 11.4" in nvidia-smi.
